scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -96,4 +96,14 @@ private[scio] object ScioUtil {
     }
   }
 
+  /**
+   * Modulo operation, This always return a non negative value in the range [0, mod)
+   * Returns incorrect value for `mod(Int.MinValue, Int.MaxValue)` as Int.MaxValue - 1
+   * but remainder of Int.MinValue % Int.MaxValue is -1
+   */
+  @inline def mod(value: Int, mod: Int): Int = {
```

Comment:
@anish749 couldn't we also consider using `Math.floorMod` since partitions can never be negative?

Reply:
I wasn't aware of that. But looks like it can be the best solution. I'll try that out and push. Thanks for pointing out 👍